File: Calculator/sci/sci.py
from flask import Flask, request, jsonify
import math

app = Flask(__name__)

@app.route('sci.html', methods=['GET'])
def calculate():
    expr = request.args.get('expr')
    try:
        # Évaluer l'expression mathématique en toute sécurité
        result = eval(expr, {"__builtins__": None}, {"sin": math.sin, "cos": math.cos, "tan": math.tan,
                                                      "log": math.log10, "sqrt": math.sqrt, "pow": math.pow,
                                                      "exp": math.exp, "pi": math.pi, "e": math

File: Calculator/sci/sci.js
// Fonction pour ajouter un chiffre ou un opérateur au champ d'affichage
function appendValue(value) {
    const result = document.getElementById('result');
    if (result.value === 'Erreur') {
        result.value = '';
    }
    result.value += value;
}

// Fonction pour vider l'affichage
function clearDisplay() {
    const result = document.getElementById('result');
    result.value = '';
}

// Fonction pour calculer l'expression et afficher le résultat
function calculate() {
    const result = document.getElementById('result');
    try {

        const expr = result.value.replace(/Math.sqrt\(/g, 'Math.pow(').replace(/Math.log\(/g, 'Math.log10(');
        result.value = eval(expr);
    } catch (e) {
        result.value = 'Erreur';
    }
}
